src/chart.ts

```diff
@@ -553,7 +553,11 @@ export class Chart extends EventEmitter.EventEmitter {
         },
         (d: any) => {
           // The key is the function set or other value that uniquely identifies the datum.
-          return d.fn || d.r || d.x || d.text
+          // Add the graphType to the key to handle graphType updates on the same function (#344)
+          return JSON.stringify({
+            function: d.fn || d.r || d.x || d.text,
```

Comment:
Now that I think about it, we could plot:

```
data: [
  {fn: 'x'},
  {fn: 'x'},
]
```

In which case the key would be `x` for both, each one should have a different key instead. I filed another bug to track this https://github.com/mauriciopoppe/function-plot/issues/346